torch_xla/csrc/xla_sharding_util.h

```diff
@@ -46,7 +46,23 @@ class ShardingUtil {
       std::vector<xla::ComputationClient::DataPtr> arguments,
       std::vector<std::string> devices);
 
-  // Shard a tensor and returns the sharded tensors which belong on `devices`
+  // Processes replicated execution results, where `sharded_results` contains
+  // `PjRtData` handles and spans the number of devices (outer) and the number
+  // of arguments (innner). This requires `sharding_specs` of the same size as
+  // the number of arguments. `sharding_specs` can contain `nullptr` if the
+  // corresponding result argument is not sharded. The replicated execution
+  // leaves the results in replicated states, which is aligned with the default
+  // exepctation `replicated_output=true`. However, if we override the
+  // compiler's default behavior and allow the execution to return sharded
+  // results, then we should set `replicated_output=false` and wrap sharded
+  // arguments into `PjRtShardedData`. This returns a vector of size that is
+  // equal to the number of arguments.
+  static std::vector<xla::ComputationClient::DataPtr> OutputHandler(
+      std::vector<std::vector<xla::ComputationClient::DataPtr>> sharded_results,
+      std::vector<XLATensor::ShardingSpecPtr> sharding_specs,
+      bool replicated_output = true);
```

Comment:
are we ever going to need `replicated_output=false`?

Reply:
Yes, when we change this,
```
// TODO(yeounoh) this is set to false by default, but explicitly set here
// to expose the knob for future reference. We can override the compiler's
// default behavior to further optimize parameter sharding in the future.
compile_options.executable_build_options
          .set_allow_spmd_sharding_propagation_to_output({false});
```